lib/diagnostic.js

```diff
@@ -8,22 +8,38 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
+  let result = arguments[0]; // if there are any arguments, 'result' equals the first one. But if not, then it returns undefined
 
+  for (let i = 0; i < arguments.length; i++) {
+    if (result > arguments[i]){
+      result = arguments[i];
+    }
+  }
+  return result;
 };
 
+
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
-const sum = function sum() {
-
+const sum = function sum(a,b,c){
+  let result = [];
+  for (let i = 0; i < array.length; i++) {
+    let array[i] = array[()(a + b + c) / 3 )];
+  }
+  return result
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
-const newArray = function newArray(/* length, defaultValue */) {
-
+const newArray = function newArray(length, defaultValue) {
+  let result = [];
+  for (let i = 0; i < array.length; i++) {
+    array[i] = defaultValue[i];
+  }
+  return result;
 };
```

Comment:
I think you might have misunderstood the question here. Please study this solution (it's for #3 and #4) and make sure you can understand this. 
```
const newArray = function newArray(length, def) {
  let defFunc = def instanceof Function ? def : () => def;
  let array = new Array(length);
  for (let i = 0; i < length; i++) {
    array[i] = defFunc(i);
  }
  return array;
};
```